elytron-oidc-client/src/main/java/org/wildfly/extension/elytron/oidc/ElytronOidcSubsystemTransformers.java

```diff
@@ -29,25 +34,31 @@ public void registerTransformers(SubsystemTransformerRegistration registration)
 
         ChainedTransformationDescriptionBuilder chainedBuilder = TransformationDescriptionBuilder.Factory.createChainedSubystemInstance(registration.getCurrentSubsystemVersion());
         // 3.0.0 (WildFly 32) to 2.0.0 (WildFly 29)
-//        from3(chainedBuilder);
+        from3(chainedBuilder);
         // 2.0.0 (WildFly 29) to 1.0.0 (WildFly 28)
         from2(chainedBuilder);
 
-        chainedBuilder.buildAndRegister(registration, new ModelVersion[] { VERSION_1_0_0.getVersion() });
+        chainedBuilder.buildAndRegister(registration, new ModelVersion[] { VERSION_2_0_0.getVersion(), VERSION_1_0_0.getVersion() });
     }
 
     private static void from2(ChainedTransformationDescriptionBuilder chainedBuilder) {
         ResourceTransformationDescriptionBuilder builder = chainedBuilder.createBuilder(VERSION_2_0_0.getVersion(), VERSION_1_0_0.getVersion());
         builder.rejectChildResource(PathElement.pathElement(SECURE_SERVER));
     }
-    // TODO: Figure out how to handle the transformer with a preview level attribute being added to the subsystem
-//    private static void from3(ChainedTransformationDescriptionBuilder chainedBuilder) {
-//        ResourceTransformationDescriptionBuilder builder = chainedBuilder.createBuilder(VERSION_3_0_0.getVersion(), VERSION_2_0_0.getVersion());
-//        builder.addChildResource(PathElement.pathElement(SECURE_SERVER))
-//                .getAttributeBuilder();
-//
-//        builder.addChildResource(PathElement.pathElement(SECURE_DEPLOYMENT))
-//                .getAttributeBuilder();
-//    }
+
+    private static void from3(ChainedTransformationDescriptionBuilder chainedBuilder) {
+        ResourceTransformationDescriptionBuilder builder = chainedBuilder.createBuilder(VERSION_3_0_0.getVersion(), VERSION_2_0_0.getVersion());
+
+        builder.addChildResource(PathElement.pathElement(SECURE_SERVER))
+                .getAttributeBuilder()
+                .addRejectCheck(RejectAttributeChecker.DEFINED, SCOPE)
+                .setDiscard(DiscardAttributeChecker.UNDEFINED, SCOPE);
+
+        builder.addChildResource(PathElement.pathElement(SECURE_DEPLOYMENT))
+                .getAttributeBuilder()
+                .addRejectCheck(RejectAttributeChecker.DEFINED, SCOPE)
+                .setDiscard(DiscardAttributeChecker.UNDEFINED, SCOPE);
+        builder.rejectChildResource(PathElement.pathElement(SCOPE));
```

Comment:
This line shouldn't be needed since `scope` isn't a child resource (it's an attribute for `secure-server` and `secure-deployment`).

Reply:
Fixed, but I think discard and rejection is not working right still, causing the tests to fail when scope is defined in the XML files and pass when it is commented out.